test/decimal128_test.dart

```diff
@@ -0,0 +1,305 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:math';
+
+import 'package:meta/meta.dart';
+import '../lib/src/decimal128.dart';
+import 'package:test/test.dart';
+
+const int defaultTimes = 100000;
+
+void repeat(dynamic Function() body, [int times = defaultTimes]) {
+  for (var i = 0; i < times; ++i) {
+    body();
+  }
+}
+
+@isTest
+void repeatTest(String description, dynamic Function(Decimal128 x, int xInt, Decimal128 y, int yInt) body, [int times = defaultTimes]) {
+  final r = Random(42); // use a fixed seed to make tests deterministic
+  test('$description ($times variations)', () {
+    repeat(
+      () {
+        // 2^31 ensures x * y doesn't overflow
+        var xInt = r.nextInt(1 << 31);
+        final x = Decimal128.fromInt(xInt);
+        var yInt = r.nextInt(1 << 31);
+        final y = Decimal128.fromInt(yInt);
+
+        body(x, xInt, y, yInt);
+      },
+      times,
+    );
+  });
+}
+
+Future<void> main([List<String>? args]) async {
+  test('Decimal128.nan', () {
```

Comment:
What about adding some test for query realm by Decimal128?

Reply:
Done